### ISSUE TITLE
'make clean' is broken for luajit tests

### ISSUE THREAD

**Issue by ligurio** [tarantool/tarantool#5520]
### Tarantool version:

5d2962244247707c7298a59de0bd4628091dda96

### OS version:

Ubuntu 20.04 x86_64

### Bug description:

`make` is unable to find Makefile generated by CMake in a source dir `libsandwich`

### Steps to reproduce:
```
- git clone https://github.com/tarantool/tarantool
- cd tarantool && mkdir build && cd build
- cmake ..
- make -j
- make clean
make[2]: third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/build.make: No such file or directory
make[2]: *** No rule to make target 'third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/build.make'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:7315: third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/clean] Error 2
make: *** [Makefile:158: clean] Error 2
- VERBOSE=1 make clean
...
cd /home/sergeyb/sources/MRG/tarantool/build/test/unit && /usr/bin/cmake -P CMakeFiles/column_mask.test.dir/cmake_clean.cmake
make[2]: Leaving directory '/home/sergeyb/sources/MRG/tarantool/build'
make -f third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/build.make third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/clean
make[2]: Entering directory '/home/sergeyb/sources/MRG/tarantool/build'
make[2]: third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/build.make: No such file or directory
make[2]: *** No rule to make target 'third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/build.make'.  Stop.
make[2]: Leaving directory '/home/sergeyb/sources/MRG/tarantool/build'
make[1]: *** [CMakeFiles/Makefile2:7315: third_party/luajit/test/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/clean] Error 2
make[1]: Leaving directory '/home/sergeyb/sources/MRG/tarantool/build'
make: *** [Makefile:158: clean] Error 2
```

The problem actually is not related to gh-4427-ffi-sandwich test, but for all subdirectories in luajit/test.
Patch below fixes a problem, but disables three regression tests:
 ```patch
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index ec37580..3056a07 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,3 +1,3 @@
-add_subdirectory(gh-4427-ffi-sandwich)
-add_subdirectory(lj-flush-on-trace)
-add_subdirectory(misclib-getmetrics-capi)
+#add_subdirectory(gh-4427-ffi-sandwich)
+#add_subdirectory(lj-flush-on-trace)
+#add_subdirectory(misclib-getmetrics-capi)
```

**Comment by igormunkin** [comment]
@ligurio, I guess the issue is resolved in scope of #4862, so I tried the scenario you shared above on the bleeding master (that is a02cfe6 at the moment). 

```
$ git clone https://github.com/tarantool/tarantool tarantool-5520
<snipped>
$ cd tarantool-5520 && mkdir -p build && cd build
$ cmake ..
<snipped>
$ make -j
<snipped>
$ make clean VERBOSE=1
<snipped>
make -f third_party/luajit/test/tarantool-tests/CMakeFiles/tarantool-tests.dir/build.make third_party/luajit/test/tarantool-tests/CMakeFiles/tarantool-tests.dir/clean
make[2]: Entering directory '/tarantool-5520/build'
cd /tarantool-5520/build/third_party/luajit/test/tarantool-tests && /usr/bin/cmake -P CMakeFiles/tarantool-tests.dir/cmake_clean.cmake
make[2]: Leaving directory '/tarantool-5520/build'
make -f third_party/luajit/test/tarantool-tests/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/build.make third_party/luajit/test/tarantool-tests/gh-4427-ffi-sandwich/CMakeFiles/libsandwich.dir/clean
make[2]: Entering directory '/tarantool-5520/build'
cd /tarantool-5520/build/third_party/luajit/test/tarantool-tests/gh-4427-ffi-sandwich && /usr/bin/cmake -P CMakeFiles/libsandwich.dir/cmake_clean.cmake
make[2]: Leaving directory '/tarantool-5520/build'
<snipped>
$ echo $?
0
```

Hence, the issue looks to be fixed via 07c83aa.

**Comment by ligurio** [comment]
`make clean` works fine on 2.9.0-165-ga02cfe60c, a02cfe60cf23f4a8853dea7cf6c728b898b4c485
